### PR TITLE
D operator compliant to orcaJS master and release

### PR DESCRIPTION
**Pull request by npisanti**
I made some tweaks to the D operator to make it work like orcaJS (i checked both the master and release version), and i also formatted with clang-format 

I hope everything is good!

### CODE REVIEW

File: sim.c [hunk 1]
```diff
@@ -1,6 +1,6 @@
+#include "sim.h"
 #include "gbuffer.h"
 #include "mark.h"
-#include "sim.h"
```
Comment (cancel) [inline sim.c]: These headers should not be reordered like this -- clang-foramt is supposed to keep the .h for a .c at the top, then alphabetically sort the other ones below it.

File: sim.c [hunk 2]
```diff
@@ -249,7 +249,7 @@ static void oper_poke_and_stun(Glyph* restrict gbuffer, Mark* restrict mbuffer,
 
 #define MOVEMENT_CASES                                                         \
   'N' : case 'n' : case 'E' : case 'e' : case 'S' : case 's' : case 'W'        \
-      : case 'w'
+                                                               : case 'w'
```
Comment (cancel) [inline sim.c]: Not sure why clang-format made this change here. Are you sure it's reading the `.clang-format` file for this project?

File: sim.c [hunk 3]
```diff
@@ -463,11 +463,13 @@ BEGIN_OPERATOR(delay)
   PORT(0, -1, IN | PARAM);
   PORT(0, 1, IN);
   PORT(1, 0, OUT);
+  Usz mod_num = index_of(PEEK(0, 1));
   Usz rate = index_of(PEEK(0, -1));
-  Usz offset = index_of(PEEK(0, 1));
```
Comment (cancel) [inline sim.c]: Prefer to read from the lower addresses first, then work up towards higher addresses -- `PEEK(0, -1)` will be at a lower address than `PEEK(0, 1)`. This is generally easier on the memory prefetcher.

File: sim.c (hunk 2 above)
Comment (npisanti) [reply sim.c]: i'm new to clang-format, i just used `clang-format -i sim.c`, from the root of Orca-c, shouldn't it work like this?

Comment (npisanti) [reply sim.c]: ok, i should have used `clang-format -style=file -i sim.c` instead, sorry

File: sim.c (hunk 3 above)
Comment (npisanti) [reply sim.c]: i just copied the code from clock, so we have to invert the PEEKs also there

File: sim.c (hunk 2 above)
Comment (cancel) [reply sim.c]: Huh. Ok, I noticed two things:
1. The existing header include order was wrong. `sim.h` is supposed to be at the top. I don't know how it became moved down.
2. I updated my version of `clang-format`, and now it wants to move `sim.h` back to the top. But it doesn't do the thing with moving the `case 'w'` to the right like that (which is obviously wrong.)

I think both of these hunks should just be removed.

File: sim.c (hunk 3 above)
Comment (cancel) [reply sim.c]: Correct, looks like they got messed up at some point.

**Pull request merged**